pkgs/by-name/ld/ld64/patches/0003-Always-use-write-instead-of-mmap.patch

```diff
@@ -0,0 +1,50 @@
+From 49489b079a3a2fd42f62c62441981cc734b0d7e7 Mon Sep 17 00:00:00 2001
+From: Randy Eckenrode <randy@largeandhighquality.com>
+Date: Wed, 13 Nov 2024 13:53:14 -0500
+Subject: [PATCH 03/20] Always use `write` instead of `mmap`
+
+This is already required on aarch64-darwin (even under Rosetta 2), so also do it on x86_64-darwin to
+avoid needing to use private headers to get the definition of `_COMM_PAGE_CPU_CAPABILITIES64`.
+---
+ src/ld/OutputFile.cpp | 18 ++----------------
+ 1 file changed, 2 insertions(+), 16 deletions(-)
+
+diff --git a/src/ld/OutputFile.cpp b/src/ld/OutputFile.cpp
+index edefb71..487b338 100644
+--- a/src/ld/OutputFile.cpp
++++ b/src/ld/OutputFile.cpp
+@@ -57,7 +57,7 @@ extern "C" {
+ 
+ #include <CommonCrypto/CommonDigest.h>
+ #include <AvailabilityMacros.h>
+-#include <System/machine/cpu_capabilities.h>
++
```

Comment:
nit: unnecessary newline?

Reply:
These three headers are set off by themselves in the file. This patch deletes one of them, preserving the structure.